riscv_analysis_cli/src/printer.rs

```diff
@@ -148,38 +149,38 @@ impl JSONPrint {
     }
 
     /// Convert a single diagnostic item to JSON
-    fn to_json<T: FileReader + Clone> (&self, parser: &RVParser<T>, item: &DiagnosticItem) -> Value {
+    fn wrap_item<T: FileReader + Clone> (&self, parser: &RVParser<T>, item: &DiagnosticItem) -> DiagnosticWrapper {
         // Get the fields
         let path = parser
             .reader
             .get_filename(item.file)
-            .map(|f| fs::canonicalize(f).unwrap_or_default());
+            .map(|f| fs::canonicalize(f).unwrap_or_default())
+            .map(|p| p.to_str().unwrap_or_default().to_string());
         let level = match item.level {
             SeverityLevel::Error => "Error",
             SeverityLevel::Warning => "Warning",
             SeverityLevel::Information => "Info",
             SeverityLevel::Hint => "Hint",
         };
 
-        // Convert to JSON
-        json!({
-            "file": path,
-            "title": item.title,
-            "description": item.description,
-            "level": level,
-            "range": {
-                "start" : self.position_to_json(item.range.start),
-                "end" : self.position_to_json(item.range.end),
+        DiagnosticWrapper {
+            file: path,
+            title: item.title.clone(),
+            description: item.description.clone(),
+            level: level.to_string(),
+            range: RangeWrapper {
+                start: self.wrap_position(item.range.start),
+                end: self.wrap_position(item.range.end),
             }
-        })
+        }
     }
 
-    fn position_to_json(&self, pos: Position) -> Value {
-        json!({
-            "line": pos.line,
-            "column": pos.column,
-            "raw": pos.raw_index,
-        })
+    fn wrap_position(&self, pos: Position) -> PositionWrapper {
+        PositionWrapper {
```

Comment:
I would implement `From<Position>` on `PositionWrapper`, same for `RangeWrapper`.

That way, this call can be simplified to
```rust
pos.into()
```

An implementation of `From` could look like this.
```rust
impl From<Position> for PositionWrapper {
  fn from(item: Position) -> Self {
    Self {
      line: item.line,
      column: item.column,
      raw: item.raw_index,
    }
  }
}
```